docs/design/hierarchical-queue-on-capacity-plugin.md

```diff
@@ -0,0 +1,118 @@
+# Hierarchical Queue on Capacity Plugin
+
+## Motivation
+
+In the context of cloud-native AI task scheduling, fair scheduling and resource utilization are two critical concerns for users. The Volcano community has already developed the elastic queue management plugin, known as capacity, to support fine-grained resource borrowing/lending and queue management, thereby improving resource utilization. However, in real-world scenarios, queues are often hierarchical, corresponding to the hierarchical organizational structure of company teams. To better align with practical queue usage scenarios and further enhance resource utilization for AI task scheduling, Volcano needs to build upon the existing capacity plugin and introduce hierarchical queue management capabilities.
+
+## In Scope
+
+- Enhancing the elastic queue management capabilities of the capacity plugin to support hierarchical queues.
+
+## Out of Scope
+
+- Extending the hierarchical queues of any other plugins.
+
+## User Stories
+
+### Story 1
+
+As an administrator, I want to create hierarchical queues to effectively manage resource allocation between different queues. This includes managing resources between parent and child queues, ensuring that the sum of the capacity/guarantee resources of the child queues does not exceed that of the parent queue.
+
+### Story 2
+
+As a user, I expect that only leaf queues can schedule jobs/podgroups. This ensures efficient management of resources and tasks at different levels of the queue hierarchy. 
+
+### Story 3
+
+When the resource status of a child queue changes, the resource status of its parent queues should be updated from the bottom up.
+
+### Story 4
+
+When executing the reclaim and preempt actions, the resource reclamation and preemption should consider the hierarchical queues. The system should prioritize queues belonging to the same parent queue as the preemptor, and then consider other queues from the bottom up in the hierarchy.
+
+For example, consider the following hierarchical queue structure and jobs:
+
+<div align="center"><img width="582" height="393" src="images/hierarchical-queue-on-capacity-plugin.png" /></div>
+
+When job A performs reclaim or preempt actions, the system should first consider job B (belonging to the same parent queue a), and then consider job C (belonging to the same parent queue root).
+
+## Design detail
+
+### Data structure modifications
+
+- Extend the queueAttr struct in capacity plugin to include two new fields:
```

Comment:
Do we have a limit on the number of sub- child?

Reply:
Currently, there is no limit. If you have any suggestions or recommendations regarding limiting the number of sub-child, I would greatly appreciate your input.